engine/src/cmd/damageable.cpp

```diff
@@ -247,15 +247,26 @@ void Damageable::ApplyDamage( const Vector &pnt,
         //static float hull_dam_to_eject    =
         //    GameConfig::GetVariable( "physics", "hull_damage_to_eject", 100.0 );
         static float auto_eject_percent =
-                GameConfig::GetVariable( "physics", "autoeject_percent", 0.5 );
+                GameConfig::GetVariable( "physics", "autoeject_percent", 0.5f );
         static bool player_autoeject =
                 GameConfig::GetVariable( "physics", "player_autoeject", true );
 
-        if(shot_at_is_player && player_autoeject) {
-            unit->EjectCargo( (unsigned int) -1 );
-        } else if(rand() < (RAND_MAX*auto_eject_percent) && unit->isUnit() == _UnitType::unit &&
-                  unit->faction != neutralfac && unit->faction != upgradesfac) {
-            unit->EjectCargo( (unsigned int) -1 );
+        if (shot_at_is_player) {
+            if (player_autoeject
+                    && rand() < (RAND_MAX * auto_eject_percent)) {
+                VS_LOG(debug, "Auto ejecting player");
+                unit->EjectCargo( (unsigned int) -1 );
+            } else {
+                VS_LOG(debug, "Not auto ejecting player");
+            }
+        } else {
+            if (unit->isUnit() == _UnitType::unit
+                    && rand() < (RAND_MAX * auto_eject_percent)) {
+                VS_LOG(debug, "Auto ejecting NPC");
+                unit->EjectCargo( (unsigned int) -1 );
+            } else {
+                VS_LOG(debug, "Not auto ejecting NPC");
+            }
```

Comment:
OK, try it now?

I tested this code just now, and I was able to confirm that all four cases were exercised:

1. Auto ejecting NPC
2. Not auto ejecting NPC
3. Auto ejecting player
4. Not auto ejecting player

Reply:
OK. This looks good and I'll approve it.

I have some minor nitpicks:

- This is very readable but I'd shorten it and lose readability for something like:
`if(unit->isUnit && ((shot_at_is_player && player_auto_eject) || !shot_at_is_player)`
   This would collapse a lot of the code into something shorter. But, like I said, it's a balance and taste issue.
- I just realized that the if (unit->isUnit() == _UnitType::unit) is potentially going to be a big issue moving forward. If we move ships to their own class as we should and is planned, we'll need to go through the code and find all these instances.

Reply:
Oops. I can't approve as I'm not a reviewer.

Reply:
I just re-requested a review from you, @royfalk . Hopefully that allows you to approve it.